architecture/2025-04-compute-platform-extensibility.md

```diff
@@ -0,0 +1,505 @@
+func fromAppExtensionClassificationDataModel(e datamodel.Extension) ExtensionClassification {
+    switch e.Kind {
+    case datamodel.KubernetesMetadata:
+        var ann, lbl = fromExtensionClassificationFields(e)
+        return &KubernetesMetadataExtension{
+            Kind:        to.Ptr(string(e.Kind)),
+            Annotations: *to.StringMapPtr(ann),
+            Labels:      *to.StringMapPtr(lbl),
+        }
+    case datamodel.KubernetesNamespaceExtension:
+        return &KubernetesNamespaceExtension{
+            Kind:      to.Ptr(string(e.Kind)),
+            Namespace: to.Ptr(e.KubernetesNamespace.Namespace),
+        }
+    case datamodel.ACIExtension:
+        return &AzureContainerInstanceExtension{
+            Kind:          to.Ptr(string(e.Kind)),
+            ResourceGroup: to.Ptr(e.AzureContainerInstance.ResourceGroup),
+        }
+    }
+
+    return nil
+}
+```
+
+```go
+pkg/corerp/frontend/controller/environments/createorupdateenvironment.go (98)
+
+    if newResource.Properties.Compute.Kind == rpv1.ACIComputeKind {
+        if err := e.createOrUpdateACIEnvironment(ctx, newResource); err != nil {
+            return nil, err
+        }
+    }
+```
+
+<!--
+High level description of updates to each component. Provide information on 
+the specific sub-components that will be updated, for example, controller, processor, renderer,
+recipe engine, driver, to name a few.
+
+#### UCP (if applicable)
+#### Bicep (if applicable)
+#### Deployment Engine (if applicable)
+#### Core RP (if applicable)
+#### Portable Resources / Recipes RP (if applicable)
+-->
+
+### Error Handling
+
+The approach to error handling will stay the same, with two additional features:
+
+## Test plan
+
+A basic test scenario will be added to Radius that will be included in the cloud functional tests.
+
+## Security
+
+The Radius [Secret Store](https://docs.radapp.io/reference/resource-schema/core-schema/secretstore/) portable type will create an Azure KeyVault when deploying to ACI.
+
+## Compatibility
+
+* When a resource type is not supported by the compute kind, a descriptive error will be emitted.
+* When a "punch through" specification is added to a bicep file to provide platform specific configuration, configurations that do not apply to the target environment will be ignored and not emit an error.
+
+## Monitoring and Logging
+
+No changes to monitoring and logging.
+
+## Development plan
+
+### Milestone 1: Add the ACI feature to Radius
+
+Exit Criteria
+
+* The ACI integration is merged with the main branch in the Radius repo.
+* A recorded presentation exists.
+* Customers can reproduce the demo on their own.
+
+### Milestone 2: Serverless Extensibility
+
+Exit Criteria
+
+* An extensibility interface that allows compute platforms to be added.
+* Documentation on how to create a compute environment.
+
+## Open Questions
+
+* Can we use Dapr for secret stores on ACI, or should we hard code KeyVault?
```

Comment:
We already have an implementation for Azure KeyVault where we can mount a keyvault as a volume on a container. I'm guessing there would be some overlap in scenarios here that might be useful for this

Reply:
Good point and I'm adding this to the question.